packages/core/utils/src/dml/modifiers/optional.ts

```diff
@@ -0,0 +1,36 @@
+import { SchemaType } from "../types"
+import { NullableModifier } from "./nullable"
+
+export class OptionalModifier<T> {
+  /**
+   * A type-only property to infer the JavScript data-type
+   * of the schema property
+   */
+  declare $dataType: T | undefined
+
+  /**
+   * The parent schema on which the nullable modifier is
+   * applied
+   */
+  #schema: SchemaType<T>
```

Comment:
Should we use `_` as in other places? Same for other private variables

Reply:
we used `_` because we were using a loose private attribute which is accessible on the prototype outside of the class. With `#` private menber they are not available and you can't conflict with a getter as it is prefixed

Reply:
Yeah, same as @adrien2p . We don't need convention here, since the field is private natively